hive/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -163,6 +171,140 @@ public void renameTable(TableIdentifier from, TableIdentifier to) {
+        return null;
+      });
+
+    } catch (AlreadyExistsException e) {
+      throw new org.apache.iceberg.exceptions.AlreadyExistsException("Namespace '%s' already exists!",
+            namespace);
+
+    } catch (TException e) {
+      throw new RuntimeException("Failed to create namespace " + namespace.toString() + " in Hive MataStore", e);
+
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException(
+          "Interrupted in call to createDatabase(name) " + namespace.toString() + " in Hive MataStore", e);
+    }
+  }
+
+  @Override
+  public List<Namespace> listNamespaces(Namespace namespace) {
+    if (namespace.levels().length > 1) {
+      throw new NoSuchNamespaceException("Namespace does not exist: %s", namespace);
+    }
+
+    try {
+      return namespace.isEmpty() ? clients.run(HiveMetaStoreClient::getAllDatabases).stream().map(Namespace::of)
+          .collect(Collectors.toList()) : Stream.of(Namespace.empty()).collect(Collectors.toList());
+    } catch (TException e) {
+      throw new RuntimeException("Failed to list all namespace: " + namespace.toString() + " in Hive MataStore",  e);
+
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException(
+          "Interrupted in call to getAllDatabases()" + namespace.toString() + " in Hive MataStore", e);
+    }
+  }
+
+  @Override
+  public boolean dropNamespace(Namespace namespace) {
+    if (namespace.levels().length != 1) {
+      throw new NoSuchNamespaceException("Namespace does not exist: %s", namespace);
+    }
+
+    try {
+      clients.run(client -> {
+        client.dropDatabase(namespace.level(0),
+            false /* deleteData */,
+            false /* ignoreUnknownDb */,
+            true /* cascade */);
+        return null;
+      });
+
+      return true;
+
+    } catch (NoSuchObjectException e) {
+      throw new NoSuchNamespaceException("Namespace %s does not exist: %s", namespace, e);
+
+    } catch (TException e) {
+      throw new RuntimeException("Failed to drop namespace " + namespace.toString() + " in Hive MataStore", e);
+
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException(
+          "Interrupted in call to drop dropDatabase(name)" + namespace.toString() + " in Hive MataStore", e);
+    }
+  }
+
+  @Override
+  public boolean setNamespaceMetadata(Namespace namespace, Map<String, String> meta) {
+    Map<String, String> currentMeta = loadNamespaceMetadata(namespace);
+    if (meta.containsKey("location")) {
+      throw new UnsupportedOperationException(
+          "Does not support change 'location' in HiveCatalog: " + namespace.toString());
+    }
+    if (meta.containsKey("name")) {
+      throw new UnsupportedOperationException(
+          "Does not support change 'name' in HiveCatalog: " + namespace.toString());
+    }
+
+    meta.forEach(currentMeta::put);
```

Comment:
This doesn't have the correct semantics. When setting namespace metadata, the map of metadata replaces the old map. Any existing keys should be overwritten or removed. Merging old metadata and new metadata should be done before calling this method.